### PR TITLE
add order to questions in a set

### PR DESCRIPTION
**Pull request by TomasNiemczyk98**
## Summary

Created a new table for many to many relation between questions and question sets which contains their order. Questions returned from repository are already ordered.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Refactoring (non-breaking change to code and structure)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/WebApi/Domain/Models/QuestionSetQuestionModel.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+{
+    public int Order { get; set; }
+
+    public int QuestionListId { get; set; }
+
+    public int InterviewQuestionId { get; set; }
+
+    public QuestionSetModel QuestionList { get; set; }
+
+    public QuestionModel InterviewQuestion { get; set; }
```
Comment (PlesnikJakub) [inline src/WebApi/Domain/Models/QuestionSetQuestionModel.cs:13]: I don't see this model being very useful, it feels overcomplicated 🤔

**Review by PlesnikJakub — CHANGES_REQUESTED** [review]
The large portion of this PR is going to be fine but there are some areas where it feels not ideal.

1. Repository - not sure if we need a separate one for this purpose, it would be just fine to transfer this responsibility to QuestionSetRepository. 

2. I think the domain should know nothing about the relationship table, it only matters to get the order of questions. Thus API is expected to return an attribute with each question as an "order" or "sortKey".  I definitely don't want to send relationship representation over to the domain.

**Comment by PlesnikJakub** [comment]
We could use some brainstorming for the design of this feature.
@fpindej  @petrspelos

File: src/WebApi/Application/Repositories/IQuestionSetQuestionRepository.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using Domain.Entities;
+using Domain.Models;
+using System.Threading.Tasks;
+
+namespace Application.Repositories;
+
+public interface IQuestionSetQuestionRepository : IGenericRepository<QuestionSetQuestionModel, IEntity>
+{
+    Task<bool> Remove(int listId, int questionId);
```
[the remark this answers is not in the document]
Comment (TomasNiemczyk98) [reply src/WebApi/Application/Repositories/IQuestionSetQuestionRepository.cs:9]: Yeah that makes sense, the model was only necessary as type parameter for generic repository with the repository being removed there is no need for it anymore

**Review by PlesnikJakub — APPROVED** [review]
ship it 🚀

**Pull request merged**